x-pack/docs/en/rest-api/security/has-privileges-user-profile.asciidoc

```diff
@@ -0,0 +1,122 @@
+
+[[security-api-has-privileges-user-profile-desc]]
+==== {api-description-title}
+
+This API is similar to the <<security-api-has-privileges>> API, which checks the privileges only of the user
+that's calling it.
+However, this API uses the profile IDs, as returned by <<security-api-activate-user-profile>>,
+to identify the users for which to check the privileges of.
+
+See <<security-privileges>> for the list of privileges that can be specified in this API.
+
+A successful call returns the subset list of profile IDs that have **all** the requested privileges.
+
+[[security-api-has-privileges-user-profile-request-body]]
+==== {api-request-body-title}
+
+`uids`:: (list) A list of <<security-api-activate-user-profile-response-body, profile IDs>> that identify
+the users for which to check the privileges of.
+
+`privileges`:: The object containing all the privileges to be checked.
+`cluster`::: (list) A list of the cluster privileges that you want to check.
+`index`:::
+`names`:::: (list) A list of indices.
+`allow_restricted_indices`::: (Boolean) This needs to be set to `true` (default
+is `false`) if using wildcards or regexps for patterns that cover restricted
+indices. Implicitly, restricted indices do not match index patterns because
+restricted indices usually have limited privileges and including them in
+pattern tests would render most such tests `false`. If restricted indices are
+explicitly included in the `names` list, privileges will be checked against
+them regardless of the value of `allow_restricted_indices`.
+`privileges`:::: (list) A list of the privileges that you want to check for the
+specified indices.
+`application`:::
+`application`:::: (string) The name of the application.
+`privileges`:::: (list) A list of the privileges that you want to check for the
+specified resources. May be either application privilege names, or the names of
+actions that are granted by those privileges.
+`resources`:::: (list) A list of resource names against which the privileges
+should be checked.
+
+Note that the `privileges` section above is identical to the
+<<security-api-has-privileges-request-body, request body of the other Has Privileges API>>.
+
+[[security-api-has-privileges-user-profile-response-body]]
+==== {api-response-body-title}
+
+A successful has privileges user profile API call returns a JSON structure that contains
+two list fields:
+
+`has_privilege_uids`:: (list) The subset of the requested profile IDs of the users that have
+**all** the requested privileges.
+
+`error_uids`:: (list) The subset of the requested profile IDs for which an error was
+encountered. It does **not** include the missing profile IDs or the profile IDs of
+the users that do not have all the requested privileges. This field is absent if empty.
+
+[[security-api-has-privileges-user-profile-example]]
+==== {api-examples-title}
+
+The following example checks whether the two users identified by the given profile IDs have all the
```

Comment:
I feel `associated` reads better. Also I think we can drop `IDs` for simplicity given the context is clear. WDYT?
```suggestion
The following example checks whether the two users associated to the given profile have all the
```